apis/dataplane/v1beta1/openstackdataplaneservice_types.go

```diff
@@ -53,6 +53,7 @@ type OpenstackDataPlaneServiceCert struct {
 	// not set, OpenStackDataPlaneService.Spec.EDPMServiceType is used. If
 	// OpenStackDataPlaneService.Spec.EDPMServiceType is not set, then
 	// OpenStackDataPlaneService.Name is used.
+	// DEPRECATED: Will be removed in a future release. Use EDPMServiceType instead.
```

Comment:
@dprince @stuggi did we ever decide on a consistent way to deprecate fields and migrate any existing CRs that exist?

Reply:
There is a way, but it would probably involve ConversionWebhooks. If you mark them as deprecated now that is fine. Then in the future we can create a new *version* of this CRD and then create a ConversionWebhook to automatically convert CRs to that upon update.